File: motomanager/frontend/motomanager-web/src/app/models/service-order.ts
export type ServiceOrderStatus = 'Open' | 'InProgress' | 'Closed';

export interface ServiceOrder {
  id: number;
  vehicleId: number;
  description: string;
  status: ServiceOrderStatus;
  date: string;
  mileage: number;
  openedAt: string;
  closedAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateServiceOrderRequest {
  vehicleId: number;
  description: string;
  date: string;
  mileage: number;
}

export interface ApplyActivityDefaultsResult {
  operationsAdded: number;
  materialsAdded: number;
}

export interface UpdateServiceOrderRequest {
  vehicleId: number;
  description: string;
  status: ServiceOrderStatus;
  date: string;
  mileage: number;
}

File: motomanager/backend/MotoManager.Domain/Entities/ServiceOrder.cs
using MotoManager.Domain.Enums;

namespace MotoManager.Domain.Entities;

public class ServiceOrder
{
    public long Id { get; set; }
    public long VehicleId { get; set; }
    public string Description { get; set; } = string.Empty;
    public ServiceOrderStatus Status { get; set; } = ServiceOrderStatus.Open;
    public DateTimeOffset OpenedAt { get; set; }
    public DateTimeOffset? ClosedAt { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public Vehicle? Vehicle { get; set; }
}
